docs/solana/options.rst

```diff
@@ -1,5 +1,202 @@
-Solana specific options / CLI flags
-====================================
+Solana-Specific Options / CLI Flags
+===================================
 
-This page documents Solana-specific Certora Prover options, these are either CLI flags or `prover_args` flags.
+This page documents Solana-specific Certora Prover options, which include CLI flags or `prover_args` flags.
 
+The `certoraSolanaProver` utility invokes the Rust compiler and then sends the job to Certora's servers.
+
+The most commonly used command is:
+
+.. code-block:: bash
+
+    certoraSolanaProver --build_script <path_to_build_script>
+
+If a precompiled execution is desired, the run command can skip the compilation step by executing:
+
+.. code-block:: bash
+
+    certoraSolanaProver <path_to_binary_file>
+
+A short summary of these options can be seen by invoking:
+
+.. code-block:: bash
+
+    certoraSolanaProver --help
+
+Using Configuration (Conf) Files
+--------------------------------
+
+For larger projects, the command line for running the Certora Prover can become large and cumbersome. It is therefore recommended to use configuration files instead.
+
+These files are in [JSON5](https://json5.org/) format and use a `.conf` extension. They hold the parameters and options for the Prover.
+
+For more details, see `Conf File <https://docs.certora.com/en/latest/docs/prover/cli/conf-file-api.html#conf-files>`.
+
+
+.. contents:: Overview
+   :depth: 2
+   :local:
+
+Modes of Operation
+------------------
+
+The Certora Solana Prover has two modes of operation, using a predefined build script, and passing precompiled binary directly.
+These modes are mutually exclusive - you cannot run the tool with more than one mode at a time.
+
+.. _build_script:
+--build_script
+~~~~~~~~~~~~~~~
+
+**What does it do?**
+
+Runs formal verification of specified properties while providing an automatic method to compile a Rust project.
+
+The build script must output the following:
+
+- `project_directory`: Path to the project root directory.
+- `sources`: List of source files or directories used or imported in the program. Source files should be relative to the `project_directory` with support of wildcards. All files declared in this list will be uploaded as sources to cloud and displayed in the rule report. Source files are required, for instance, for the jump to source feature to work.
+- `executables`: List of compiled binary files, which are the target of the Rust program.
+- `success`: Boolean flag indicating if the build phase passed successfully.
+
```

Comment:
@yoav-el-certora 
Would be useful to have a subsection with exact specifications for the build script.
What I missing from the above:
1. build script must support `--json` flag
2. build script should output nothing or json (depending on the flag)
3. build script should output 0 on success and 1 on failure
4. exact format of the json output
5. build script may optionally provide paths to files for stdout and stderr of the build
6. path to inline and summary files